peru/main.py

```diff
@@ -221,6 +221,59 @@ def do_override(params):
             print('{}: {}'.format(module, params.runtime.get_override(module)))
 
 
+@peru_command('module', '''\
+Usage:
+    peru module add [<module>] [<url>] [<type>] [--rev=<rev>] [--reup=<reup>] 
+             [--filename=<filename>] [--sha=<sha>] [--unpack=<unpack>] [--yaml=<yaml>]
+    peru module --help
+
+Add a module to your Peru yaml file. Reguires a module name, url, and module type.
+
+Options:
+    add                       add new module
+      --rev=<rev>             set module rev value
+      --reup=<reup>           set module reup value
+      --filename=<filename>   set module filemame value
+      --sha=<sha>             set module sha1 value
+      --unpack=<unpack>       set module unpack value 
+      --yaml=<yaml>           file to use instead of peru.yaml     
+      --help 
+''')
+def do_add(params):
+    if params.args['add']:
+        if not params.args['<module>']:
+            params.runtime.display.print('Please specify a module to add. See usage with peru module --help')
+        elif not params.args['<url>']:
+            params.runtime.display.print('Please specify a module url. See usage with peru module --help')
+        elif not params.args['<type>']:
+            params.runtime.display.print('Please specify a module type. See usage with peru module --help')
```

Comment:
Instead of doing all the work to check required fields in code, just make them required in the docopt string above, by removing the surrounding `[ ]` braces. Docopt is really nice for this.